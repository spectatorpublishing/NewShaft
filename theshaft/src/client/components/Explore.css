.DormButton {
	text-align: left;
	border: 5px solid #9B9B9B;
	color: #9B9B9B;
	display: inline-block;
	cursor: pointer;
}

div.pics {
	margin:3em;
	display: inline-block;
}

img {
	padding: 5px;
}

img.blinky {

}

img.pinky {

}

img.inky {

}

img.clyde {

}