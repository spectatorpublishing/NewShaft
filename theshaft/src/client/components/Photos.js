import React, { Component } from "react";
import styled from 'styled-components';

let Imagecontainer = styled.div`
  display: flex;
  flex-direction: row;
  height: 40vh;
  width: 100%;
  overflow: hidden;
  align-items: center;
`

let PicsBig = styled.div`
  display: flex;
  flex-direction: column;
  width: 66%;
  height: 100%;
  overflow: hidden;
  align-items: center;
`

let PicsMid = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  width: 33%;
  overflow: hidden;
`

let PicsLow = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
  height: 50%;
  overflow: hidden;
`

let Pinky = styled.img`
  height: 50%;
`

let Inky = styled.img`
  height: 100%;
  width: 50%;
`

let Clyde = styled.img`
  height: 100%;
  width: 50%;
`


export default class Photos extends Component {
  constructor(props) {
    super(props);

    this.state = {
      imageOne: this.props.imageOne,
      imageTwo: this.props.imageTwo,
      imageThree: this.props.imageThree,
      imageFour: this.props.imageFour
    };
  }

  render() {
    return (
      <Imagecontainer>
        <PicsBig>
          <img src={this.state.imageOne} />
        </PicsBig>
        <PicsMid>
          <Pinky src={this.state.imageTwo} />
          <PicsLow>
            <Inky src={this.state.imageThree} />
            <Clyde src={this.state.imageFour} />
          </PicsLow>
        </PicsMid> 
      </Imagecontainer>
    );
  }
}
