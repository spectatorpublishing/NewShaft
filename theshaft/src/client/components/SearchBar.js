import React, { Component } from 'react';
import styled from 'styled-components'

let Input = styled.input`
  border-radius: 10px;
  border: 1px solid ${props => props.theme.mediumGray};
  display: block;
  width: 80%;
  font-family: inherit;
  font-size: 1rem;
  margin: 0.75rem 10%;
  background: white;
  line-height: 1.5rem;
  padding: 0 5%;
  box-sizing: border-box;
  @media only screen and (min-width: 768px) {
	border-radius: 0;
    border: none;
    border-bottom: 1px solid ${props => props.theme.lightGray};
    display: block;
    width: 80%;
    font-family: inherit;
    font-size: 1rem;
    font-style: italic;
    margin: 0.75rem 0;
    padding: 0;
    box-sizing: content-box;
  }
`

export default class SearchBar extends Component {

	constructor(props){
		super(props)
		this.searchChange = this.searchChange.bind(this)
	}
	
	searchChange(e){
		this.props.handleChange(e.target.value, "Dorm")
	}
	  
	render(){
		return <Input
                type="text"
                onChange={this.searchChange}
                placeholder="Search for your new home..."
              />;
	}
}