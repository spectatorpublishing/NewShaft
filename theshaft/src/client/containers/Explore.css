.header {
	margin: 0 auto;
	width: 100%;
	text-align: center;
}

.SchoolButton {
	font-size: 20pt;
	text-align: center;
	border: 1px solid #9B9B9B;
	color: #9B9B9B;
	display: inline-block;
	height: 50px;
	margin: 20px;
	padding: 5px;
	cursor: pointer;
}

.DormButtons {
	margin: 0 auto;
	width: 100%;
}
