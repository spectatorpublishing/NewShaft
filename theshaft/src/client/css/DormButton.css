.DormButton {
	text-align: left;
	border: 5px solid #9B9B9B;
	color: #9B9B9B;
    display: inline-block;
	cursor: pointer;
    margin: 10px;
}
