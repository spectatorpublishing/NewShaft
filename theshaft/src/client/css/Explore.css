div.explore {
    width: 100%;
    height: 100%;
    padding: 0 auto;
}

div.sidebar div.map {
    padding: 0 auto;
}

div.sidebar {
    width: 33%;
    min-height: 100%;
    left: 0;
    right: 0;
    top: 0;
    z-index: 1;
    overflow: hidden;
}

div.map {
    position: fixed;
    float: right;
    width: 67%;
    right: 0;
    top: 0;
    z-index:1;
}
