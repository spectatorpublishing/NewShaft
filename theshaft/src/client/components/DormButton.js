import React, { Component } from 'react';
import './DormButton.css';

export default class DormButton extends Component {

	constructor(props) {
		super(props);

		this.state = {
			name: this.props.name,
			address: this.props.address,
			sundial_distance: this.props.sundial_distance,
			description: this.props.description
		}
	}

	render() {
		return (
			<div className="DormButton" onClick={() => this.onClick(this.state.name)}>
				<h3> {this.state.name} </h3>
				<h3> {this.state.address} </h3>
				<h3> {this.state.sundial_distance} </h3>
				<h3> {this.state.description} </h3>
				<br/>
				<h2> CLICK TO VIEW FLOOR PLAN </h2>
			</div>
		)
	}
}
