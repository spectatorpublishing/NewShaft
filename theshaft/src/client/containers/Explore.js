import React, { Component } from 'react';
import styled from "styled-components";
import { Route, Link, BrowserRouter as Router } from 'react-router-dom';
import DormButton from '../components/DormButton';
import '../css/Explore.css';
import Dorm from './Dorm.js';
import Updater from '../components/Updater';
import ExploreSidebar from "../components/ExploreSidebar";
//import "../css/Explore.css";
import map from "../assets/map.png";
import Maps from "../components/Maps";

// div.explore {
//   width: 100%;
//   height: 100%;
//   padding: 0 auto;
//   overflow: hidden;
//   /* flex-direction: row; */
// }

let Explore = styled.div`
  width: 100%;
  height: 100%;
  padding: 0 auto;
  overflow: hidden;
  flex-direction: row;
`

let SideBar = styled.div`
  width: 90%;
  padding-left: 1em;
  overflow-y: scroll; 
  @media only screen and (min-width: 768px) {
    width: 30%;
    min-height: 100%;
    z-index: 1;
  }
`

let MapView = styled.div`
display: none;
width: 0%;
@media only screen and (min-width: 768px) {
  display: inline;
  position: fixed;
  padding-left: 1em;
  float: right;
  width: 67%;
  right: 0;
  top: 0;
  z-index:1;
}
`

let ColOne = styled.div`
  display: flex;
  width: 100%;
`

let ColTwo = styled.div`
  display: inline;
  position: fixed;
  padding-left: 1em;
  float: right;
  // width: 67%;
  right: 0;
  top: 0;
  z-index:1;
  // display: flex;
  //flex-direction: column;
  width: ${(mobile) => mobile ? `67%`: `50%`};
  //width: 50%;
`

export default class Explore extends Component {
  constructor(props){
    super(props);
    fetch('/api/filterDorm', {
      method: 'POST',
      headers: {
        'X-Powered-By': 'Express',
        'Content-Type': 'application/json; charset=utf-8'
      },
      body: {
        "college": -1,
        "single": true,
        "double": true,
        "triple": true,
        "suite": [],
        "make_up":[]
      }
    }).then((res) => {console.log("\nRESPONSE: "); console.log(JSON.stringify(res));})
      .then((dorms) => {
        this.setState({
          dorms: dorms.map((dorm) => {
            return {
              id: dorm['dorm'],
              school: dorm['college'],
              name: dorm['dorm'] + 'Hall',
              image: dorm['thumbnail_image'],
              description: dorm['description'],
              amenities: dorm['amenities']
            };
          })
        });
      });
  }
  
  render() {
    // var dorms = [
    //   {
    //     id: "McBain",
    //     school: "Columbia",
    //     name: "McBain Hall",
    //     image: "https://housing.columbia.edu/files/housing/McBain.jpg",
    //     description:
    //       "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nulla nulla, condimentum a mattis in, faucibus id sapien. Sed rhoncus.",
    //     amenities: "No AC"
    //   },
    //   {
    //     id: "Carman",
    //     school: "Columbia",
    //     name: "Carman Hall",
    //     image: "https://housing.columbia.edu/files/housing/Carman.jpg",
    //     description:
    //       "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nulla nulla, condimentum a mattis in, faucibus id sapien. Sed rhoncus.",
    //     amenities: "No AC"
    //   },
    //   {
    //     id: "Sulzberger",
    //     school: "Barnard",
    //     name: "Sulzberger Tower",
    //     image: "https://housing.columbia.edu/files/housing/McBain.jpg",
    //     description:
    //       "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nulla nulla, condimentum a mattis in, faucibus id sapien. Sed rhoncus.",
    //     amenities: "No AC"
    //   },
    //   {
    //     id: "mcbain",
    //     school: "Columbia",
    //     name: "McBain Hall",
    //     image: "https://housing.columbia.edu/files/housing/McBain.jpg",
    //     description:
    //       "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nulla nulla, condimentum a mattis in, faucibus id sapien. Sed rhoncus.",
    //     amenities: "No AC"
    //   },
    //   {
    //     id: "mcbain",
    //     school: "Columbia",
    //     name: "McBain Hall",
    //     image: "https://housing.columbia.edu/files/housing/McBain.jpg",
    //     description:
    //       "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla nulla nulla, condimentum a mattis in, faucibus id sapien. Sed rhoncus.",
    //     amenities: "No AC"
    //   }
    // ];


    return (

      <Explore>
        <ColOne>
          <SideBar>
            <ExploreSidebar dorms={dorms} />
          </SideBar>
        </ColOne>
        <ColTwo>
          <Maps latitudes={[40.7128, 40.7129, 40.7128]} longitudes={[-74.006, -74.007, -74.008]} popupInfo={["Carman", "McBain", "John Jay"]} popupId={["Carman", "McBain", "JohnJay"]}/>
        {/* <div className="map">
            <img src={map} />
        </div> */}
        </ColTwo>
        <Updater interval="15" />
      </Explore>
    );
  }
}
