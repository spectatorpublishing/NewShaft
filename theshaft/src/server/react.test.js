import React from 'react';
import Photos from '../client/components/Photos.js';
import SearchBar from '../client/components/SearchBar.js';
import DormButton from '../client/components/DormButton.js';
import Explore from '../client/containers/Explore.js';
import renderer from 'react-test-renderer';

it('Photos renders correctly', () => {
  const tree = renderer
    .create(<Photos image="http://www.facebook.com"></Photos>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('SearchBar renders correctly', () => {
  const tree = renderer
    .create(<SearchBar/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('DormButton renders correctly', () => {
  const tree = renderer
    .create(<DormButton name="ADI House" address="21 Savage St." sundial_distance="12 minutes" description="It's lit"/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Explore renders correctly', () => {
  const tree = renderer
    .create(<Explore/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});