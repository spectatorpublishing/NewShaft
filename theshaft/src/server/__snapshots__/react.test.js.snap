// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos renders correctly 1`] = `
<div class="picsHigh">
        <img class="blinky" src={this.state.imageOne} />
        <div class="picsMid">
          <img class="pinky" src={this.state.imageTwo} />
          <div class="picsLow">
            <img class="inky" src={this.state.imageThree} />
            <img class="clyde" src={this.state.imageFour} />
          </div>
        </div>
      </div>
`;

exports[`SearchBar renders correctly 1`] = `
<div>
  <input
    className="searchbar textEntry"
    type="text"
  />
  <input
    className="searchbar submitButton"
    src={
      Object {
        "process": [Function],
      }
    }
    type="image"
  />
</div>
`;
